### PR TITLE
PYIC-2791 added dns to core-front dev env

### PR DESCRIPTION
**Pull request by vam-gds**
### What changed
PYIC-2791 added dns to core-front dev env

### Why did it change
PYIC-2791 added dns to core-front dev env

### Issue tracking

- [PYIC-2791](https://govukverify.atlassian.net/browse/PYIC-2791)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-2791]: https://govukverify.atlassian.net/browse/PYIC-2791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by TobySaundersGDS — DISMISSED** [review]

File: deploy/template.yaml [hunk 1]
```diff
@@ -128,6 +128,51 @@ Mappings:
...
+      DomainNameConfigurations:
+        - CertificateArn: !Ref DevSSLCert
+          EndpointType: REGIONAL
+          SecurityPolicy: TLS_1_2
+
+  DevApiMapping:
+    Type: AWS::ApiGatewayV2::ApiMapping
+    Condition: IsNewDevelopment
+    Properties:
+      DomainName: !Sub "${Environment}.dev.identity.account.gov.uk"
```
Comment (Wynndow) [inline deploy/template.yaml]:
```suggestion
      DomainName: !Ref DevApiDomain
```

File: deploy/template.yaml [hunk 2]
```diff
@@ -128,6 +128,51 @@ Mappings:
...
+      - DevApiDomain
+      - APIStageDefault
+
+  # dns rcord
+  DevDNSRecord:
+    Type: AWS::Route53::RecordSet
+    Condition: IsNewDevelopment
+    Properties:
+      Type: A
+      Name: !Sub "${Environment}.dev.identity.account.gov.uk"
```
Comment (Wynndow) [inline deploy/template.yaml]:
```suggestion
      Name: !Ref DevApiDomain
```

File: deploy/template.yaml [hunk 3]
```diff
@@ -128,6 +128,51 @@ Mappings:
 
 Resources:
 
+  # ssl cert
+  DevSSLCert:
+    Type: AWS::CertificateManager::Certificate
+    Condition: IsNewDevelopment
+    Properties:
+      DomainName: !Sub "${Environment}.dev.identity.account.gov.uk"
```
Comment (Wynndow) [inline deploy/template.yaml:136]:
```suggestion
      DomainName: !Ref DevApiDomain
```

File: deploy/template.yaml [hunk 4]
```diff
@@ -128,6 +128,51 @@ Mappings:
...
+      DomainValidationOptions:
+        - DomainName: !Sub "${Environment}.dev.identity.account.gov.uk"
```
Comment (Wynndow) [inline deploy/template.yaml:138]:
```suggestion
        - DomainName: !Ref DevApiDomain
```

**Review by Wynndow — COMMENTED** [review]
Feel free to ignore this, functionally the same

File: deploy/template.yaml [hunk 5]
```diff
@@ -128,6 +128,51 @@ Mappings:
...
+          HostedZoneId: !ImportValue DevIdentityHostedZoneId
+      ValidationMethod: DNS
+
+  # api domain entries / mapping
+  DevApiDomain:
+    Type: AWS::ApiGatewayV2::DomainName
+    Condition: IsNewDevelopment
+    Properties:
+      DomainName: !Sub "${Environment}.dev.identity.account.gov.uk"
```
Comment (Wynndow) [inline deploy/template.yaml:147]: Wondering if this should be `"$core-front-{Environment}.dev.identity.account.gov.uk"` or `"$core-{Environment}.dev.identity.account.gov.uk"` just to be super obvious when running journeys what you're looking at.

Comment (Wynndow) [reply deploy/template.yaml:147]: Discussed offline that it's fine as is. Should be clear from context what you're looking at.

**Review by Wynndow — APPROVED** [review]

**Pull request merged**